src/com/sap/piper/integration/TransportManagementService.groovy

```diff
@@ -68,20 +68,39 @@ class TransportManagementService implements Serializable {
 
         def proxy = config.proxy ? config.proxy : script.env.HTTP_PROXY
 
-        script.sh """#!/bin/sh -e
-                curl ${proxy ? '--proxy ' + proxy + ' ' : ''} -H 'Authorization: Bearer ${token}' -F 'file=@${file}' -F 'namedUser=${namedUser}' -o responseFileUpload.txt  --fail '${url}/v2/files/upload'
+        def responseFileUpload = 'responseFileUpload.txt'
+
+        def responseContent
+
+        try {
+            script.sh """#!/bin/sh -e
+                curl ${proxy ? '--proxy ' + proxy + ' ' : ''} -H 'Authorization: Bearer ${token}' -F 'file=@${file}' -F 'namedUser=${namedUser}' -o ${responseFileUpload}  --fail '${url}/v2/files/upload'
             """
 
-        def responseContent = script.readFile("responseFileUpload.txt")
+            echo("File upload successful.")
 
-        if (config.verbose) {
-            echo("${responseContent}")
-        }
+        } catch (Exception e) {
+            String message = 'Exception caught during file upload.'
+            if (config.verbose) {
+                message += ' Consider re-running in verbose mode in order to get more details.'
+            }
+            echo(message)
+            throw e
+        } finally {
 
-        echo("File upload successful.")
+            if (script.fileExists(responseFileUpload)) {
+                responseContent = script.readFile(responseFileUpload)
 
+                if (config.verbose) {
+                    echo("Upload file response: ${responseContent}")
+                }
+            }
+        }
+        if (! responseContent) {
+            // should not happen. Instead there should be an exception thrown by the sh step
+            error "Cannot provide upload file response."
```

Comment:
Where this "error" object is taken from?

Reply:
see [here](https://jenkins.io/doc/pipeline/steps/workflow-basic-steps/#error-error-signal)

Reply:
Should not it be script.error then?

Reply:
yes, true. Should be `script.error`

Reply:
Then also the echo above should be script.echo, right?

Reply:
yes. I will check this for all the changed coding. We are not in a step here, hence we have to use `script`.

Reply:
> Then also all the echos should be script.echo

They are. See method `private echo(message)` that is used in the class.